source/includes/_advanced-usage.md

```diff
@@ -104,67 +104,138 @@ __To open the checkout view for a product directly:__
 
-### Building the state
+Server-side rendering takes place in several steps:
 
-We assume that you know ids of products, collections and you know if you want to render Shop component on the specific url.
-Let's say that following four variables are defined: `POS_ID`, `PRODUCT_IDS`, `COLLECTION_IDS` and `IS_SHOP_ON_PAGE`.  
-Please note that `PRODUCT_IDS`, `COLLECTION_IDS` and `IS_SHOP_ON_PAGE` should depend on URL - different pages can have different products/collections.
-To build the state you should use our `StateBuilder` class. You can do it in the following way:
+- building a state containing the data needed for rendering
+
+- passing the state to the `SsrTipserElementsProvider`
+
+- rendering the application
+
+- sending the same state to the browser and passing it back to the `SsrTipserElementsProvider` on the client side.
+
+
+### Basic method
+In the basic method, you should use the `StateBuilder` class to build the state. Start with creating its instances (which can be global):
 
 ```typescript
 import { StateBuilder } from '@tipser/tipser-elements';
 
 const stateBuilder = new StateBuilder(POS_ID);
 ```
 
-Now you can use it in your request handler: 
+Then, while handling a specific request, use the `stateBuilder.buildState` method, passing an object containing information about which elements are placed on the website, i.e. what products (PRODUCT_IDS), collections (COLLECTION_IDS) and whether there is the shop component (IS_SHOP_ON_PAGE).
 
 ```typescript
-stateBuilder.buildState(PRODUCT_IDS, COLLECTION_IDS, IS_SHOP_ON_PAGE): Promise<TipserState>
+stateBuilder.buildState({
+    productIds: PRODUCT_IDS,
+    collectionIds: COLLECTION_IDS,
+    shouldFetchStore: IS_SHOP_ON_PAGE,
+})
 ```
 
-Full example:
+This method returns a promise in which the ready state will be available, which should then be passed to `SsrTipserElementsProvider`:
 
 ```typescript
-stateBuilder
-  .buildState(PRODUCT_IDS, COLLECTION_IDS, IS_SHOP_ON_PAGE)
-  .then(initialState => {
-    renderToString(
-      <TipserElementsProvider posId={POS_ID} initialState={initialState}>
-        <YourAppHere />
-      </TipserElementsProvider>
+stateBuilder.buildState({
+    productIds: dataToFetch.productIds,
+    collectionIds: dataToFetch.collectionIds,
+    shouldFetchStore: dataToFetch.shouldFetchStore
+}).then((initialState) => {
+    const markup = renderToString(
+        <TipserElementsProvider posId={POS_ID}>
+            <SsrTipserElementsProvider initialState={initialState}>
+                <App/>
+            </SsrTipserElementsProvider>
+        </TipserElementsProvider>
     );
-  });
+    ...
+ });
 ```
-
-The state should be transferred to the frontend app. You can use the pattern that is known from Redux based apps.
-All you need to do is including the following line in your html response:
+The status should then be transferred to the browser. For this purpose, it can be assigned to the global window object and placed in the returned html document:
```

Comment:
~The status~ The state,

 but we could call it also `initialState` so its clear about what exactly value are we talking about.